app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,39 @@
+class VotesController < ApplicationController
+
+
+  def show
+  end
+
+  def new
+    @vote = Vote.new
+  end
+
+  def create
+    @work = Work.find_by(id: params[:work_id])
+    @user = User.find_by(id: session[:user_id])
+
+    if @user.nil?
+      flash[:error] = "Please log in to upvote media."
+      return
+      redirect_back(fallback_location: root_path)
```

Comment:
You need to do the return **after** you redirect them.

```suggestion
      redirect_back(fallback_location: root_path)
      return
```